test/promise/evil-promises.js

```diff
@@ -20,7 +29,7 @@ describe('Evil promises should not be able to break invariants', function () {
     };
 
     var calledAlready = false;
-    Promise.resolve(evilPromise).then(function (value) {
+    new Promise(function (r) { r(evilPromise); }).then(function (value) {
```

Comment:
Hmm - `Promise.resolve` should have the same behavior as using `resolve` from `new Promise`, shouldn't it?